pkg/exchange/ftx/stream.go

```diff
@@ -102,15 +117,69 @@ func (s *Stream) SetPublicOnly() {
 	atomic.StoreInt32(&s.publicOnly, 1)
 }
 
-func (s *Stream) Subscribe(channel types.Channel, symbol string, _ types.SubscribeOptions) {
-	if channel != types.BookChannel {
+func (s *Stream) Subscribe(channel types.Channel, symbol string, option types.SubscribeOptions) {
+	if channel == types.BookChannel {
+		s.addSubscription(websocketRequest{
+			Operation: subscribe,
+			Channel:   orderBookChannel,
+			Market:    toLocalSymbol(TrimUpperString(symbol)),
+		})
+
+	} else if channel == types.KLineChannel {
+		// FTX does not support kline channel, do polling
+		go s.subscribeKLine(symbol, option)
+	} else {
 		panic("only support book channel now")
 	}
-	s.addSubscription(websocketRequest{
-		Operation: subscribe,
-		Channel:   orderBookChannel,
-		Market:    toLocalSymbol(TrimUpperString(symbol)),
-	})
+}
+
+func (s *Stream) handleChannelKlineMessage() {
+	for {
+		kline := <-s.klineMessage
```

Comment:
use select { } clause to avoid channel blocking and add context deadline checking here.

Reply:
fixed